jaqpotpy/models/sklearn.py

```diff
@@ -181,15 +186,44 @@ def _create_onnx(self, onnx_options: Optional[Dict] = None):
                         self._map_onnx_dtype(self.dataset.X[feature].dtype.name),
                     )
                 )
-
-        self.onnx_model = convert_sklearn(
+        if 'XGB' in type(self.model).__name__:
+            if self.task == "BINARY_CLASSIFICATION" or self.task == "MULTICLASS_CLASSIFICATION":
+                self._convert_classifier()
+            elif self.task == "REGRESSION":
+                self._convert_regressor()
+            else:
+                raise ValueError("Task not supported for XGBoost models")
+            self.onnx_model = to_onnx(
+            self.trained_model,
+            initial_types=self.initial_types,
+            name=name,
+            target_opset={"": 15, "ai.onnx.ml": 1}
+        )
+        else:    
+            self.onnx_model = convert_sklearn(
```

Comment:
Currently, the section where we check if XGBoost and then handle it separately from the typical sklearn flow makes future extensions harder (e.g., if we later want to add support for more model types more than xgboost and sklearn).

A better approach could be abstracting out the model-specific logic (such as in _convert_classifier and _convert_regressor) into a separate class. This way, adding new models (or extending existing ones) wouldn't require modifying the core logic each time. It would also make the code more modular and maintainable.

For example you can create a model class called XGBoost that inherits from this `SklearnModel` 
so keep SKLearn method like it was before and add a convert_model_to_onnx()
```suggestion
        convert_model_to_onnx()
        
        def convert_model_to_onnx(): 
             # sklearn same code as before
```

and create an extra class that inherits from the SKLearnModel, uses the same methods, but also has 
```python
class XGBoostModel(SklearnModel):
   convert_to_onnx(): 
       # xgboost specific code here, that converts the model
    
    # also define here those 2 methods that are only for the xgboost model 
    def _convert_regressor(self):
        update_registered_converter(
            model=XGBRegressor,
            alias="XGBRegressor",
            shape_fct=calculate_linear_regressor_output_shapes,
            convert_fct=convert_xgboost,
        )

    def _convert_classifier(self):
        update_registered_converter(
            model=XGBClassifier,
            alias="XGBClassifier",
            shape_fct=calculate_linear_classifier_output_shapes,
            convert_fct=convert_xgboost,
            options={"nocl": [True, False], "zipmap": [True, False, "columns"]})
```